ecosystem/sep-0006.md

```diff
@@ -59,7 +59,11 @@ The JWT should be included in all requests as request header:
 Authorization: Bearer <JWT>
 ```
 
-It is highly recommended to require authentication for all endpoints other than `/info` (and optionally `/fee`).
+It is highly recommended to require authentication for all endpoints other than `/info` (and optionally `/fee`). Some functionality, including the [Shared Accounts](#shared-accounts) functionality described in the section below, may only be possible through the use of SEP-10.
+
+### Shared Accounts
+
+Client applications can use a single Stellar account to hold multiple users' funds. To differentiate users using the same Stellar account, the decoded SEP-10 JWT may contain `memo` and `memo_type` attributes. The anchor should use these attributes in addition to the Stellar account in their data model to represent unique users. See the [SEP-10 Memos](sep-0010.md#memos) section for more information.
```

Comment:
Ah.. I hadn't thought about the transition from this system to muxed accounts.

The problem I see with taking this approach is the transition for users of shared accounts that have already been registered with anchor services using other memo types via request parameters. 

In SEP-31, memos will continue to be specified as request parameters, so we don't have the issue there, assuming sending anchors will never use muxed accounts to identify their customers, and I don't see why they would.

But for SEP-6 and SEP-24 clients and anchors, they're using text and hash memo types if the account is shared, and would be expected to use those memos in the JWTs with this change. I think the question is, how many users are using shared accounts with anchor services today? And even if the number is small, are we willing to cause a compatibility issue for them in order to more easily migrate to a new system in the future?

Curious on your thoughts here.

Reply:
Beyond muxed accounts, I think that removing `memo_type` will simplify these changes quite a bit

> how many users are using shared accounts with anchor services today?

I might not understand this question but how can there be users using pooled accounts today without support for that in the seps?

Reply:
The changes in this PR specify how memos can be used to narrow the scope of authenticated sessions for users of a shared account, but using shared accounts have been possible with our SEPs for some time, although there are some semantic issues that this PR resolves.

For example, a service using shared accounts for their users could integrate with an anchor for withdrawals of their stellar funds simply by specifying the `memo` and `memo_type` request parameters in the `/withdraw` request. These parameters were [defined](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0024.md#request-1) for this exact purpose:

The definition for memo:

> (optional) A wallet will send this to uniquely identify a user if the wallet has multiple users sharing one Stellar account. The anchor can use this along with account to look up the user's KYC info.

So the capability existed. But it may make sense to make the breaking change now when the user base is small compared to allowing different memo types now and then requiring anchor services and client applications to migrate to id-based memos when muxed accounts come around.